### ISSUE TITLE
"Search with DuckDuckGo" in private window works only if "Show autocomplete in address bar" is enabled

### ISSUE THREAD

**Issue by Tonev** [brave/brave-browser#17556]
## Description 

The `Search with DuckDuckGo` option in private window works only if the `Show autocomplete in address bar` option is enabled. If the `Show autocomplete in address bar` option is disabled, all searches in private window go through user's default search engine, regardless of the `Search with DuckDuckGo` preference.

Additionally, the `Learn more` link next to the `Search with DuckDuckGo` option redirects to a page where users are asked to sign in to Brave Software.

## Steps to Reproduce

   1. Go to `brave://settings/appearance`.
   2. Disable `Show autocomplete in address bar`.
   3. Open a private window and enable `Search with DuckDuckGo`, and perform a search.

## Actual result:

The search will go through the default search engine.

## Expected result:
The search should go through `DuckDuckGo`.

## Reproduces how often: 

Easily

## Brave version (brave://version info)

<html>
<body>

Brave | 1.28.105 Chromium: 92.0.4515.131 (Official Build) (64-bit)
-- | --
Revision | 6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}
OS | Windows 7 Service Pack 1 (Build 7601)
JavaScript | V8 9.2.230.22
User Agent | Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36
Command Line | "C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe" --enable-dom-distiller --disable-domain-reliability --no-pings --origin-trial-public-key=bYUKPJoPnCxeNvu72j4EmPuK7tr1PAC7SHh8ld9Mw3E=,fMS4mpO6buLQ/QMd+zJmxzty/VQ6B1EUZqoCU04zoRU= --sync-url=https://sync-v2.brave.com/v2 --lso-url=https://no-thanks.invalid --variations-server-url=https://variations.brave.com/seed --enable-features=LegacyTLSEnforced,WebUIDarkMode,PrefetchPrivacyChanges,ReducedReferrerGranularity,PasswordImport,AutoupgradeMixedContent,SafetyTip,WinrtGeolocationImplementation --disable-features=AutofillServerCommunication,FledgeInterestGroups,HandwritingRecognitionWebPlatformApi,FlocIdComputedEventLogging,HandwritingRecognitionWebPlatformApiFinch,InterestCohortFeaturePolicy,NotificationTriggers,WebOTP,IdleDetection,PrivacySandboxSettings,FederatedLearningOfCohorts,TextFragmentAnchor,EnableProfilePickerOnStartup,SubresourceWebBundles,NetworkTimeServiceQuerying,LangClientHintHeader,LiveCaption,DirectSockets,TrustTokens,SignedExchangeSubresourcePrefetch,EnablePasswordsAccountStorage,FledgeInterestGroupAPI,AutofillEnableAccountWalletStorage,InterestCohortAPIOriginTrial,FirstPartySets,SignedExchangePrefetchCacheForNavigations --flag-switches-begin --enable-features=LegacyTLSEnforced,WebUIDarkMode,PrefetchPrivacyChanges,ReducedReferrerGranularity,PasswordImport,AutoupgradeMixedContent,SafetyTip,WinrtGeolocationImplementation,EphemeralStorage --flag-switches-end
Executable Path | C:\Program Files\BraveSoftware\Brave-Browser\Application\brave.exe
Profile Path | C:\Users\User\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default
Variations | AdRewardsStudy:NextPaymentDayEphemeralStorageStudy:EnabledNativeCosmeticFilteringStudy:EnabledPermissionLifetimeReleaseStudy:EnabledSpeedreaderReleaseStudy:Disabled

</body>
</html>

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

**Comment by SarveshMD** [comment]
This is something important. Please look after this issue.

**Comment by SarveshMD** [comment]
I still wonder how you found that disabling "Show autocomplete in address bar" causes the issue 😅✌️

**Comment by Tonev** [comment]
I accidentally found out `Search with DuckDuckGo` wasn't working properly in my daily user profile while testing various scenarios for a user in Brave Community - https://community.brave.com/t/private-window-search-defaults-to-ddg-even-when-another-search-engine-is-set-as-default/275775.

I created another user profile so I could determine whether the issue was related to my daily user profile in some way, and it turned out there was something wrong with my daily user profile. So, in order to find out what was wrong with my daily user profile, I simply began changing Brave's default settings in the new user profile until the `Search with DuckDuckGo` option broke.
